### ISSUE TITLE
tsgo does not auto-include visible @types packages when compilerOptions.types is omitted

### ISSUE THREAD

**Issue by brielov** [oxc-project/tsgolint#1054]
### Summary

In a fresh Bun project, `oxlint --type-check` / `tsgo` reports `TS2584: Cannot find name 'console'` even though stable TypeScript does not report a diagnostic for the same `tsconfig.json`.

The divergence appears to be that stable TypeScript automatically includes visible `node_modules/@types/*` packages when `compilerOptions.types` is omitted, while `tsgo` does not. In this project, explicitly passing `--types bun` makes `tsgo` pass.

### Reproduction

Start from a fresh `bun init` project with `@types/bun`, `oxlint`, and `oxlint-tsgolint` installed.

`index.ts`:

```ts
console.log("Hello via Bun!");
```

`tsconfig.json`:

```jsonc
{
  "compilerOptions": {
    "lib": ["ESNext"],
    "target": "ESNext",
    "module": "Preserve",
    "moduleDetection": "force",
    "jsx": "react-jsx",
    "allowJs": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "verbatimModuleSyntax": true,
    "noEmit": true,

    "strict": true,
    "skipLibCheck": true,
    "noFallthroughCasesInSwitch": true,
    "noUncheckedIndexedAccess": true,
    "noImplicitOverride": true,

    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "noPropertyAccessFromIndexSignature": false
  }
}
```

`.oxlintrc.json`:

```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["typescript", "unicorn", "oxc"],
  "categories": {
    "correctness": "error"
  },
  "rules": {
    "typescript/no-deprecated": "error"
  },
  "env": {
    "builtin": true
  },
  "options": {
    "typeAware": true,
    "typeCheck": true
  }
}
```

### Actual behavior

`bunx oxlint` reports:

```text
index.ts:1:1: error typescript(TS2584): Cannot find name 'console'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.
```

The same behavior reproduces with standalone `tsgo`:

```sh
bunx tsgo --noEmit --pretty false
```

```text
index.ts(1,1): error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.
```

`tsgo --listFilesOnly` only includes TypeScript built-in libs and `index.ts`; it does not include any visible ambient type packages from `node_modules/@types`.

### Expected behavior

Stable TypeScript reports no diagnostic:

```sh
bunx tsc --noEmit --pretty false
```

exits successfully.

Stable TypeScript also includes the visible ambient packages automatically:

```sh
bunx tsc --noEmit --pretty false --listFilesOnly
```

includes, among other files:

```text
node_modules/@types/bun/index.d.ts
node_modules/bun-types/index.d.ts
node_modules/@types/node/web-globals/console.d.ts
```

So this does not appear to be a missing `lib: ["dom"]` case. The `console` global is available through the automatically included Bun/Node ambient types.

Also, this passes with `tsgo` when the type package is named explicitly:

```sh
bunx tsgo --noEmit --pretty false --types bun
```

### Environment

```text
Bun: 1.3.11
oxlint: 1.72.0
oxlint-tsgolint: 0.24.0
@types/bun: 1.3.14
stable TypeScript: 5.9.3
@typescript/native-preview / tsgo: 7.0.0-dev.20260704.1
```

**Comment by brielov** [comment]
Closing after checking the TypeScript 6/7 migration docs.

This appears to be expected behavior for TypeScript 6/7 rather than a tsgolint-specific bug. TypeScript 6 changes the default `compilerOptions.types` behavior from automatically enumerating visible `node_modules/@types/*` packages to an empty array. Since `tsgolint` targets TypeScript 7, it is using the new behavior, while stable TypeScript 5.9 still uses the old auto-inclusion behavior.

References:
- TypeScript 6.0 beta: `types` now defaults to `[]`: https://devblogs.microsoft.com/typescript/announcing-typescript-6-0-beta/#types-now-defaults-to
- Bun TS 6/7 docs: https://bun.com/docs/typescript-6

For this repro, adding `"types": ["bun"]` or temporarily `"types": ["*"]` matches the TS6/7 migration path.
